go/libraries/doltcore/sqle/dfunctions/dolt_branch.go

```diff
@@ -196,6 +214,42 @@ func validateBranchNotActiveInAnySession(ctx *sql.Context, branchName string) er
 	})
 }
 
+// parseRevisionDatabaseName parses the database or branch-qualified database name and returns the
+// database name and any specified branch name.
+func parseRevisionDatabaseName(revisionDbName string) (string, string) {
+	parts := strings.SplitN(revisionDbName, "/", 2)
+	if len(parts) == 0 {
+		return "", ""
+	} else if len(parts) == 1 {
+		return parts[0], ""
+	} else {
+		return parts[0], parts[1]
+	}
+}
+
+// removeBranchRevisionDatabase updates database provider information to ensure the specified
+// revision DB name is not tracked as a valid database.
+func removeBranchRevisionDatabase(ctx *sql.Context, revisionDbName string) error {
+	dsess, ok := ctx.Session.(*dsess.DoltSession)
+	if !ok {
+		return fmt.Errorf("unexpected session type: %T", ctx.Session)
+	}
+
+	provider := dsess.Session.Provider()
+	if provider, ok := provider.(sql.MutableDatabaseProvider); ok {
+		if provider.HasDatabase(ctx, revisionDbName) == false {
+			return nil
+		}
+
+		err := provider.DropDatabase(ctx, revisionDbName)
```

Comment:
Maybe should be different from the SQL interface method for the above reason

Reply:
I like that idea. I moved this code to a new method: `RevisionDbProvider.DropRevisionDb`